graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -182,13 +185,24 @@ public SchemaSource insert(String namespace, String newContents) {
     return new SchemaSource(namespace, newVersion, newContents);
   }
 
-  private void ensureTableExists(String namespace) throws Exception {
+  private void ensureTableExists() throws Exception {
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .keyspace(KEYSPACE_NAME)
+                .ifNotExists()
+                .withReplication(Replication.simpleStrategy(1))
```

Comment:
I don't think the persistence API exposes any way to inspect the cluster topology. We go for the most conservative strategy that works on any cluster, users can alter the keyspace to adjust the settings later. Or ideally they would create the keyspace beforehand. This will be mentioned in the docs.